chain/network/src/peer/peer_actor.rs

```diff
@@ -719,135 +742,147 @@ impl PeerActor {
         // decides whether to accept the connection or not: ctx.wait makes
         // the actor event loop poll on the future until it completes before
         // processing any other events.
-        ctx.wait(wrap_future({
+        if self.registration_buffered_actions.is_some() {
+            panic!("registration_buffered_actions should be None when starting registration");
+        }
+        self.registration_buffered_actions = Some(Vec::new());
+        self.handle.spawn("register peer", {
             let network_state = self.network_state.clone();
             let clock = self.clock.clone();
-            let conn = conn.clone();
-            async move { network_state.register(&clock,edge,conn).await }
-        })
-            .map(move |res, act: &mut PeerActor, ctx| {
-                match res {
-                    Ok(()) => {
-                        act.peer_info = Some(peer_info).into();
-                        act.peer_status = PeerStatus::Ready(conn.clone());
-                        // Respond to handshake if it's inbound and connection was consolidated.
-                        if act.peer_type == PeerType::Inbound {
-                            act.send_handshake(HandshakeSpec{
-                                peer_id: handshake.sender_peer_id.clone(),
-                                tier,
-                                protocol_version: handshake.protocol_version,
-                                partial_edge_info: partial_edge_info,
-                            });
-                        }
-                        // TIER1 is strictly reserved for BFT consensus messages,
-                        // so all kinds of periodical syncs happen only on TIER2 connections.
-                        if tier==tcp::Tier::T2 {
-                            // Trigger a full accounts data sync periodically.
-                            // Note that AccountsData is used to establish TIER1 network,
-                            // it is broadcasted over TIER2 network. This is a bootstrapping
-                            // mechanism, because TIER2 is established before TIER1.
-                            //
-                            // TODO(gprusak): consider whether it wouldn't be more uniform to just
-                            // send full sync from both sides of the connection independently. Or
-                            // perhaps make the full sync request a separate message which doesn't
-                            // carry the accounts_data at all.
-                            if conn.peer_type == PeerType::Outbound {
-                                ctx.spawn(wrap_future({
+            let handle = self.handle.clone();
+            async move {
```

Comment:
For my education, what does this do? 

Opinion: The highly nested and overly lengthy methods are hard to read and annoying. It's an existing problem so I'm not asking you to change anything here, just complaining.